manifests/conffile.pp

```diff
@@ -50,13 +50,16 @@
   Enum['present', 'absent', 'directory'] $ensure    = 'present',
   Variant[Array[String], String, Undef]  $source    = undef,
   Optional[String]                       $content   = undef,
-  Stdlib::Absolutepath                   $path      = "/etc/postfix/${name}",
+  Optional[Stdlib::Absolutepath]         $path      = undef,
   String                                 $mode      = '0640',
   Hash                                   $options   = {},
   Boolean                                $show_diff = true,
 ) {
+  include postfix
   include ::postfix::params
```

Comment:
If we include `postfix`, then we don't need to include `postfix::params` manually anymore (same applies in various locations of this PR).

I suspect though that including `postfix` might change the behavior in some cases...

Reply:
One way to avoid breaking this behavior would be to only expose `confdir` in `postfix::params`, and get it from there.

Reply:
i a concerned that it looks like a step in the "wrong" direction…  I mean, usage of `params.pp` has some limitations that Hiera tries to address but the module kind of has a different use of `params.pp` than what was "usual" by referencing `$postfix::params::foo` in various places, which is in turn causing trouble.

So, are you sure that instead of improving the actual situation you want to dig deeper the hole we are already in and would like to escape?

Reply:
I agree that going towards `params` is not a great idea. In that case, just get rid of the `postfix::params` class altogether in this class.